libraries/botframework-config/src/botConfiguration.ts

```diff
@@ -0,0 +1,230 @@
+        }
+    }
+
+    // encrypt all values in the config
+    public encrypt(secret: string) {
+        this.validateSecretKey(secret);
+
+        for (let service of this.services) {
+            (<ConnectedService>service).encrypt(secret);
+        }
+    }
+
+    // decrypt all values in the config
+    public decrypt(secret?: string) {
+        this.validateSecretKey(secret);
+
+        for (let service of this.services) {
+            (<ConnectedService>service).decrypt(secret);
+        }
+    }
+
+
+    // remove service by name or id
+    public disconnectServiceByNameOrId(nameOrId: string): IConnectedService {
+        let svs = new List<IConnectedService>(this.services);
+
+        for (let i = 0; i < svs.count(); i++) {
+            let service = svs.elementAt(i);
+            if (service.id == nameOrId || service.name == nameOrId) {
+                svs.removeAt(i);
+                this.services = svs.toArray();
+                return service;
+            }
+        }
+        throw new Error(`a service with id or name of [${nameOrId}] was not found`);
+    }
+
+    // remove a service
+    public disconnectService(type: string, id: string): void {
+        let svs = new List<IConnectedService>(this.services);
+
+        for (let i = 0; i < svs.count(); i++) {
+            let service = svs.elementAt(i);
+            if (service.type == type && service.id == id) {
+                svs.removeAt(i);
+                this.services = svs.toArray();
+                return;
+            }
+        }
+    }
+
+
+    // make sure secret is correct by decrypting the secretKey with it
+    public validateSecretKey(secret: string): void {
+        if (!secret || secret.length == 0) {
+            throw new Error('You are attempting to perform an operation which needs access to the secret and --secret is missing');
+        }
+
+        try {
+            if (!this.secretKey || this.secretKey.length == 0) {
+                // if no key, create a guid and enrypt that to use as secret validator
+                this.secretKey = encryptString(uuid(), secret);
+            } else {
+                // validate we can decrypt the secretKey, this tells us we have the correct secret for the rest of the file.
+                decryptString(this.secretKey, secret);
+            }
+        } catch {
+            throw new Error('You are attempting to perform an operation which needs access to the secret and --secret is incorrect.');
+        }
+    }
+
+
+    private legacyDecrypt(encryptedValue: string, secret: string): string {
+        // LEGACY for pre standardized SHA256 encryption, this uses some undocumented nodejs MD5 hash internally and is deprecated
+        const decipher = crypto.createDecipher('aes192', secret);
+        let value = decipher.update(encryptedValue, 'hex', 'utf8');
+        value += decipher.final('utf8');
+        return value;
+    }
+
+    public static serviceFromJSON(service: IConnectedService): ConnectedService {
```

Comment:
Since this static is a member of a node-only class, consider moving this to a utility class that can be imported and used exclusively.

Reply:
Scratch that. I see now that the data models are also node-exclusive. Is this what we want? The emulator uses these data models on both the client and node side. There are also scenarios I can come up with where these would be useful in a web-based bot. Would it make sense to move encryption/decryption out of the data models so they can be agnostic to the platform?